examples/1_Santanilla_Nanomole_HTE/example_Santanilla.ipynb

```diff
@@ -0,0 +1,8234 @@
+       "  'Nucleophile_9': {'name': 'phenol S10', 'SMILES': 'placeholder'},\n",
+       "  'Nucleophile_10': {'name': 'thiophenol S11', 'SMILES': 'placeholder'},\n",
+       "  'Nucleophile_11': {'name': 'phosphine S12', 'SMILES': 'placeholder'},\n",
+       "  'Nucleophile_12': {'name': 'boronate S14/water', 'SMILES': 'placeholder'},\n",
+       "  'Nucleophile_13': {'name': 'malonate S13', 'SMILES': 'placeholder'},\n",
+       "  'Nucleophile_14': {'name': 'alkene S15', 'SMILES': 'placeholder'},\n",
+       "  'Nucleophile_15': {'name': 'alkyne S16', 'SMILES': 'placeholder'}},\n",
+       " 'Catalysts': {'Catalyst_0': {'name': 'BINAP Pd G3 30',\n",
+       "   'SMILES': 'placeholder'},\n",
+       "  'Catalyst_1': {'name': 'DPPF Pd G3 31', 'SMILES': 'placeholder'},\n",
+       "  'Catalyst_2': {'name': 'XantPhos Pd G2 32', 'SMILES': 'placeholder'},\n",
+       "  'Catalyst_3': {'name': 'tBu3P Pd G2 33', 'SMILES': 'placeholder'},\n",
+       "  'Catalyst_4': {'name': 'PPA Pd G3 34', 'SMILES': 'placeholder'},\n",
+       "  'Catalyst_5': {'name': 'Aphos Pd G3 35', 'SMILES': 'placeholder'},\n",
+       "  'Catalyst_6': {'name': 'Xphos Pd G3 36', 'SMILES': 'placeholder'},\n",
+       "  'Catalyst_7': {'name': 'RuPhos Pd G2 37', 'SMILES': 'placeholder'},\n",
+       "  'Catalyst_8': {'name': 'DTBPF Pd G3 38', 'SMILES': 'placeholder'},\n",
+       "  'Catalyst_9': {'name': 'J009  Pd G3 39', 'SMILES': 'placeholder'},\n",
+       "  'Catalyst_10': {'name': 'MorDalPhos Pd G3 40', 'SMILES': 'placeholder'},\n",
+       "  'Catalyst_11': {'name': 'BrettPhos Pd G3 41', 'SMILES': 'placeholder'},\n",
+       "  'Catalyst_12': {'name': 'tBuXPhos Pd G3  42', 'SMILES': 'placeholder'},\n",
+       "  'Catalyst_13': {'name': 'tBuBrettPhos Pd G3 43', 'SMILES': 'placeholder'},\n",
+       "  'Catalyst_14': {'name': 'RockPhos Pd G3 44', 'SMILES': 'placeholder'},\n",
+       "  'Catalyst_15': {'name': 'AdBrettPhos Pd G3 45', 'SMILES': 'placeholder'}},\n",
+       " 'Bases': {'Base_0': {'name': 'DBU 24', 'SMILES': 'placeholder'},\n",
+       "  'Base_1': {'name': 'MTBD 25', 'SMILES': 'placeholder'},\n",
+       "  'Base_2': {'name': 'BTMG 26', 'SMILES': 'placeholder'},\n",
+       "  'Base_3': {'name': 'BEMP 27', 'SMILES': 'placeholder'},\n",
+       "  'Base_4': {'name': 'BTTP 28', 'SMILES': 'placeholder'},\n",
+       "  'Base_5': {'name': 'P2Et 29', 'SMILES': 'placeholder'}}}"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Extract unique values for each reaction component\n",
+    "reagent_dict = {}\n",
+    "component_types = ['Electrophile', 'Nucleophile', 'Catalyst', 'Base']\n",
+    "\n",
+    "# Create dictionary for each component set with names and SMILES\n",
+    "for component in component_types:\n",
+    "    component_dict = {}\n",
+    "    for i, unique in enumerate(list(data[component].unique())):\n",
```

Comment:
I don't think I understand the reason for including the indices here. Why not just use `unique` as the key?

Reply:
Cleaned this up. Was using the index to name the molecules by their reaction role, but this isn't necessary. Replaced with `unique` (i.e., the dataset name) as suggested.